diff --git a/node_modules/@rn-native-utils/workmanager/android/.classpath b/node_modules/@rn-native-utils/workmanager/android/.classpath
new file mode 100644
index 0000000..bbe97e5
--- /dev/null
+++ b/node_modules/@rn-native-utils/workmanager/android/.classpath
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17/"/>
+	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer"/>
+	<classpathentry kind="output" path="bin/default"/>
+</classpath>
diff --git a/node_modules/@rn-native-utils/workmanager/android/.project b/node_modules/@rn-native-utils/workmanager/android/.project
new file mode 100644
index 0000000..d9a88ef
--- /dev/null
+++ b/node_modules/@rn-native-utils/workmanager/android/.project
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>rn-native-utils_workmanager</name>
+	<comment>Project rn-native-utils_workmanager created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1780497793260</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/@rn-native-utils/workmanager/android/build.gradle b/node_modules/@rn-native-utils/workmanager/android/build.gradle
index f221490..99e2f96 100644
--- a/node_modules/@rn-native-utils/workmanager/android/build.gradle
+++ b/node_modules/@rn-native-utils/workmanager/android/build.gradle
@@ -1,41 +1,18 @@
-buildscript {
-  // Buildscript is evaluated before everything else so we can't use getExtOrDefault
-  def kotlin_version = rootProject.ext.has("kotlinVersion") ? rootProject.ext.get("kotlinVersion") : project.properties["BackgroundSync_kotlinVersion"]
-
-  repositories {
-    google()
-    mavenCentral()
-  }
-
-  dependencies {
-    classpath "com.android.tools.build:gradle:7.2.1"
-    // noinspection DifferentKotlinGradleVersion
-    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-  }
-}
-
 def reactNativeArchitectures() {
   def value = rootProject.getProperties().get("reactNativeArchitectures")
   return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
 }
 
-def isNewArchitectureEnabled() {
-  return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
-}
-
 apply plugin: "com.android.library"
 apply plugin: "kotlin-android"
 
-if (isNewArchitectureEnabled()) {
-  apply plugin: "com.facebook.react"
-}
-
 def getExtOrDefault(name) {
   return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties["BackgroundSync_" + name]
 }
 
 def getExtOrIntegerDefault(name) {
-  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties["BackgroundSync_" + name]).toInteger()
+  def value = rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties["BackgroundSync_" + name]
+  return value != null ? (value instanceof Number ? value : value.toString().toInteger()) : 0
 }
 
 def supportsNamespace() {
@@ -58,21 +35,21 @@ android {
     }
   }
 
-  compileSdkVersion getExtOrIntegerDefault("compileSdkVersion")
+  compileSdk getExtOrIntegerDefault("compileSdkVersion")
 
   defaultConfig {
-    minSdkVersion getExtOrIntegerDefault("minSdkVersion")
-    targetSdkVersion getExtOrIntegerDefault("targetSdkVersion")
-
+    minSdk getExtOrIntegerDefault("minSdkVersion")
+    targetSdk getExtOrIntegerDefault("targetSdkVersion")
   }
 
   buildTypes {
+    debug {}
     release {
       minifyEnabled false
     }
   }
 
-  lintOptions {
+  lint {
     disable "GradleCompatible"
   }
 
@@ -87,16 +64,12 @@ repositories {
   google()
 }
 
-def kotlin_version = getExtOrDefault("kotlinVersion")
+def kotlin_version = getExtOrDefault("kotlinVersion") ?: "2.1.20"
 
 dependencies {
-  // For < 0.71, this will be from the local maven repo
-  // For > 0.71, this will be replaced by `com.facebook.react:react-android:$version` by react gradle plugin
-  //noinspection GradleDynamicVersion
-  implementation "com.facebook.react:react-native:+"
+  implementation "com.facebook.react:react-android"
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
   implementation 'androidx.work:work-runtime-ktx:2.9.1'
   implementation "androidx.concurrent:concurrent-futures:1.2.0"
 
 }
-
diff --git a/node_modules/@rn-native-utils/workmanager/android/src/main/java/com/backgroundsync/worker/BackgroundWorker.kt b/node_modules/@rn-native-utils/workmanager/android/src/main/java/com/backgroundsync/worker/BackgroundWorker.kt
index b2ec210..ddefb98 100644
--- a/node_modules/@rn-native-utils/workmanager/android/src/main/java/com/backgroundsync/worker/BackgroundWorker.kt
+++ b/node_modules/@rn-native-utils/workmanager/android/src/main/java/com/backgroundsync/worker/BackgroundWorker.kt
@@ -13,7 +13,7 @@ class BackgroundSyncWorker(context: Context, workerParams: WorkerParameters) :
   HeadlessWorker(context, workerParams) {
   override fun getTaskConfig(data: Data?): HeadlessJsTaskConfig? {
     Log.d("TAG", "getTaskConfig: $data")
-    val taskKey= data?.getString("taskKey")
+    val taskKey= data?.getString("taskKey") ?: "defaultSyncTask"
     val maxRetryAttempts = data?.getInt("maxRetryAttempts", 3)
     val retryDelay = data?.getInt("retryDelay", 4000)
     val taskTimeout = data?.getLong("taskTimeout", 10000L)
